src/components/common/forms/SignUpForm.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { ErrorMessage } from "@hookform/error-message";
+import { useNavigate } from "react-router-dom";
+
+import { useAppSelector, useAppDispatch } from "@/state/app/hooks";
+import { setUsername } from "@/state/features/user";
+import { Button, Input, Dropdown } from "@components/common";
+import { signUpRequest } from "@/services";
+import { urlFormat } from "@/utils";
+import { Pages } from "@/pages";
+
+import styles from "./Forms.module.css";
+
+const SignUpForm = () => {
+  const { genders } = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitted, isValid },
+    setError,
+    clearErrors,
+  } = useForm();
+
+  const navigate = useNavigate();
+  const isFormValid = !isSubmitted || isValid;
+
+  const handleErrorMessage = (() => {
+    if (errors?.password?.type === "minLength") {
+      return (
+        <ErrorMessage
+          errors={errors}
+          name="password"
+          message="Password should be at least 6 characters long"
+          render={({ message }) => <>{message}</>}
+        />
+      );
+    } else if (!errors?.serverError) {
+      return "All fields are required";
+    }
+  })();
```

Comment:
Did you have the opportunity to work with schema validators? I think in this cases there are better to implement all the validations by ourselves.

Reply:
I did! I will create a new ticket to include validations through `yup`